TCAT/FavoritesTableViewController.swift

```diff
@@ -0,0 +1,138 @@
+        navigationController?.navigationBar.titleTextAttributes = titleAttributes
+        navigationController?.navigationBar.tintColor = .black
+        navigationController?.navigationBar.barTintColor = .white
+        navigationController?.navigationBar.isTranslucent = false
+        navigationController?.navigationBar.setBackgroundImage(UIImage(), for: .default)
+        navigationController?.navigationBar.shadowImage = UIImage()
+
+        allStops = FetchBusStops.shared.getAllStops()
+        tableView.register(BusStopCell.self, forCellReuseIdentifier: "BusStop")
+        tableView.register(SearchResultsCell.self, forCellReuseIdentifier: "SearchResults")
+
+        resultsSection = Section(type: .searchResults, items: [ItemType]())
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    @objc func dismissVC() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return resultsSection.items.isEmpty ? allStops.count : resultsSection.items.count
+    }
+
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let searchBar = UISearchBar()
+        searchBar.isTranslucent = true
+        searchBar.backgroundImage = UIImage()
+        searchBar.alpha = 1.0
+        let textFieldInsideSearchBar = searchBar.value(forKey: "searchField") as? UITextField
+        textFieldInsideSearchBar?.backgroundColor = .tableBackgroundColor
+        textFieldInsideSearchBar?.attributedPlaceholder = NSAttributedString(string: "Search", attributes: [NSAttributedStringKey.foregroundColor: UIColor.searchBarPlaceholderTextColor])
+       searchBar.backgroundColor = .white
+        searchBar.delegate = self
+        return searchBar
+
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell: UITableViewCell!
+        if resultsSection.items.isEmpty {
+         let cell = tableView.dequeueReusableCell(withIdentifier: "BusStop", for: indexPath) as! BusStopCell
+        cell.textLabel?.text = allStops[indexPath.row].name
+        return cell
+        }
+
+        let item = resultsSection.items[indexPath.row]
+
+        switch item {
+        case .busStop(let busStop):
+            cell = tableView.dequeueReusableCell(withIdentifier: "BusStop", for: indexPath) as! BusStopCell
+            cell.textLabel?.text = busStop.name
+        case .placeResult(let placeResult):
+            cell = tableView.dequeueReusableCell(withIdentifier: "SearchResults", for: indexPath) as! SearchResultsCell
```

Comment:
consider using a `let busStopCellIdentifier = ...` instance variable instead of putting the same string when you register and dequeue the cell